File: backend/TrainingApp/models/post.js
// models/Post.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const postSchema = new Schema({
  user: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  content: {
    type: String,
    required: true
  },
  image: {
    type: String,
    default: null
  },
  workoutDetails: {
    type: {
      type: String
    },
    duration: {
      type: Number,
      default: 0
    },
    calories: {
      type: Number,
      default: 0
    }
  },
  likes: [{
    type: Schema.Types.ObjectId,
    ref: 'User'
  }],
  comments: [{
    user: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    text: {
      type: String,
      required: true
    },
    date: {
      type: Date,
      default: Date.now
    }
  }],
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Post', postSchema);

File: android/app/src/main/java/com/trainly/app/domain/models/Post.kt
package com.trainly.app.domain.models
data class Post(val id: String, val userId: String, val userName: String, val userAvatar: String?=null, val content: String?=null, val image: String?=null, val likes: List<String>, val comments: List<Comment>, val createdAt: String, val workout: Workout?=null, val privacy: String?=null)
data class Comment(val id: String, val userId: String, val userName: String, val userAvatar: String?=null, val text: String, val createdAt: String)
